packages/ui/src/common/hooks/useFirstObservableValue.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useMemo, useState } from 'react'
+import { firstValueFrom, Observable } from 'rxjs'
+
+export function useFirstObservableValue<T>(observableFactory: () => Observable<T> | undefined, deps: readonly any[]) {
+  const [data, setData] = useState<T | undefined>(undefined)
+
+  const observable = useMemo(observableFactory, deps)
+
+  useEffect(() => {
+    if (observable) {
+      firstValueFrom(observable).then((value) => setData(value))
+    }
+  }, [observable])
+
+  return useMemo(() => data, [JSON.stringify(data)])
```

Comment:
Since only one value will be emited, I don't think the `useMemo` is useful here:
```suggestion
  return data
```